### ISSUE TITLE
Preparation to release ReaR 2.0

### ISSUE THREAD

**Issue by gdha** [rear/rear#1073]
#### Relax-and-Recover (rear) Issue Template

Please fill in the following items before submitting a new issue (quick response is not guaranteed with free support):

* rear version (/usr/sbin/rear -V): 2.0 - not yet released of course
* Brief description of the issue

Placeholder issue to link important issues that  we should finish before we release ReaR 2.0. Also, documentation updates etc...

**Comment by jsmeix** [comment]
Fixed 'ReaR' versus 'rear' spelling via
https://github.com/rear/rear/pull/1087 cf.
https://github.com/rear/rear/issues/1086 and
https://github.com/rear/rear/issues/1033

**Comment by jsmeix** [comment]
My currently known (minor) behavioural changes:

1.)
Since I implemeted real incremental backups
BACKUP_TYPE=incremantal
does no longer mean a differential backup
but now it means real incremental backups.
For real differential backup the new
BACKUP_TYPE=differential
must now be set.
See https://github.com/rear/rear/pull/1071
and https://github.com/rear/rear/issues/1069
and https://github.com/rear/rear/issues/974

2.)
ReaR default logfile name changes, see
https://github.com/rear/rear/issues/1092
2a)
The logfile that gets copied to the NETFS location
had hardcoded name "rear.log" but now it respects
the LOGFILE setting so that now it is by default named
"rear-HOSTNAME.log" according to default.conf
LOGFILE="$LOG_DIR/rear-$HOSTNAME.log"
2b)
The partial logfile that gets copied to initramfs
had hardcoded name "rear-partial-..." but now
it respects the LOGFILE setting so that now it is
by default named "rear-HOSTNAME-partial-..."
using $( basename $LOGFILE ) except a trailing '.log'
as name prefix with the default.conf setting
LOGFILE="$LOG_DIR/rear-$HOSTNAME.log"

3.)
The BugError message has changed format.
Now it is more informative because it shows the
source file where BugError or BugIfError was called
and it is less "shouting at the poor reader" because
"BUG BUG BUG! ..." is replaced by "BUG ...", see
https://github.com/rear/rear/pull/1075

**Comment by gdha** [comment]
@jsmeix @gozora Can we agree that we freeze adding **new** features into ReaR as of today? Otherwise, we will never be able to deliver v2.0 by the end of this year of beginning next year. Remember, the milestone is Fosdem 2017.

PS: bug fixes and what was already in process of being added can continue, but try to get it in before Christmas or otherwise, post-pone for v2.1.

**Comment by gozora** [comment]
Hello @gdha,

No problem for me.

**Comment by jsmeix** [comment]
Of course no new features or bigger changes from me.
All I like to get for ReaR 2.0 is that
multiple (optionally simultaneous) backups and restores
work sufficiently well.

**Comment by jsmeix** [comment]
One more (minor) behavioural change, cf.
https://github.com/rear/rear/issues/1073#issuecomment-264127724

4.)
Since https://github.com/rear/rear/pull/1130 merged
when rear ends there is now in syslog (e.g. /var/log/messages)
either "rear $WORKFLOW finished with zero exit code"
or "rear $WORKFLOW failed with exit code $EXIT_CODE".
Before it was only "DONE: rc=0" - there was no longer
a syslog message for non-zero exit codes before, cf.
https://github.com/rear/rear/issues/564 and
https://github.com/rear/rear/commit/e57893983cb29808866fcfee3b96056e0e851c57

**Comment by jsmeix** [comment]
Addendum to
https://github.com/rear/rear/issues/1073#issuecomment-267555036
see
https://github.com/rear/rear/issues/564#issuecomment-268210019

4a)
Since https://github.com/rear/rear/pull/1133 merged
when the checklayout workflow ends with exit code 1
there is now the special syslog message
<pre>
rear checklayout finished with exit code 1 (layout or config changed)
</pre>

**Comment by jsmeix** [comment]
A possibly major change that may cause
real regressions for some users:

5.)
All numbered ReaR scripts have been renumbered
from two digit prefix numbers NM_script_name.sh
to three digit prefix numbers NM0_script_name.sh
by ading a trailing 0 so that 12 becomes 120
except 00 which becomes 005.
This renumbering will cause regressions for users who have
their own 2-digit scripts because now only 3-digit scripts will
work (see the SourceStage function in framework-functions.sh)
so that 2-digit user scripts must be renamed to 3-digit names.

**Comment by jsmeix** [comment]
As far as I see from my point of view
I think ReaR 2.0 can be released now.

**Comment by jsmeix** [comment]
I noticed inconsistent documentation about what
distributions are supported by ReaR in
doc/rear-release-notes.txt
doc/user-guide/01-introduction.adoc
doc/user-guide/02-getting-started.adoc
I will do a pull request to align them...

**Comment by gdha** [comment]
@jsmeix @gozora I'm starting with the release notes for ReaR v2.00

**Comment by jsmeix** [comment]
Now it looks perfectly o.k. again.

**Comment by jsmeix** [comment]
@gdha 
only a side note FYI regarding your
https://github.com/rear/rear/issues/1073#issuecomment-270301959
"starting with the release notes"
and the matching
<pre>
git log --format="* %s : %b %n" --topo-order
</pre>
on https://github.com/rear/rear/wiki/Release-process

Persoally I use meanwhile very often something like
<pre>
git log --format="%ae %ad%n%s :%n%b%n" --graph | fmt -120 -t | less
</pre>
(where --graph implies --topo-order see "git help log")
to get a relatively easily comprehensible history.

For example
<pre>
git log --format="* %s : %b %n" --topo-order | fmt -78 -t | less
</pre>
results (excerpt)
<pre>
* Merge pull request #1148 from dl5rcw/master : Bareos support: add missing
   directory /var/run/bareos in recovery system

* add missing directory :

* Merge pull request #1147 ...
</pre>
while
<pre>
git log --format="%ae %ad%n%s :%n%b%n" --graph | fmt -78 -t | less
</pre>
shows that excerpt as
<pre>
|
*   gratien.dhaese@gmail.com Tue Jan 3 16:08:42 2017 +0100
|\  Merge pull request #1148 from dl5rcw/master :
| | Bareos support: add missing directory /var/run/bareos in recovery system
| |
| * info@dl5rcw.de Tue Jan 3 15:07:07 2017 +0100
|/  add missing directory :
|
|
*   jsmeix@suse.com Tue Jan 3 12:46:46 2017 +0100
|\  Merge pull request #1147 from
</pre>
which makes it more clear that the meaningless
"add missing directory" entry can be simply ignored
for the release notes because it is clear that this one
is the predecessor of the more meaningful
"Bareos support: add missing directory
/var/run/bareos in recovery system" entry.

**Comment by gdha** [comment]
As ReaR has been released yesterday (albeit under radar) we can close this issue. Still need to do some announcements.
